### PR TITLE
Add support of AllWinner H3 SoC - OrangePi Plus2E

### PR DESCRIPTION
**Pull request by vgrebenschikov**
Corresponding u-boot port required - sysutils/u-boot-orangepi-plus-2e
see https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=214729

### CODE REVIEW

File: board/OrangePi-Plus2E/overlay/etc/rc.conf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+hostname="orangepi-plus2e"
+ifconfig_awg0="DHCP"
+sshd_enable="YES"
+
+# If the SD card has more than 7G free, drop
+# a 768M swap file on it the first time we boot.
+#makeswapfile_enable="YES"
+#makeswapfile_minfreeblocks=14000000
+#makeswapfile_swapsize_mb=768
```
Comment (vgrebenschikov) [inline board/OrangePi-Plus2E/overlay/etc/rc.conf]: removed

File: board/OrangePi-Plus2E/overlay/etc/rc.d/makeswapfile [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
```
Comment (kientzle) [inline board/OrangePi-Plus2E/overlay/etc/rc.d/makeswapfile]: You should also remove this file.

File: board/OrangePi-Plus2E/overlay/boot/loader.rc.local [hunk 3]
```diff
@@ -0,0 +1,2 @@
+load kernel
+load -t dtb boot/dtb/orangepi-plus-2e.dtb
```
Comment (evadot) [inline board/OrangePi-Plus2E/overlay/boot/loader.rc.local]: This shouldn't be needed.
ubldr load the dtb after the uboot variable "fdtfile"

File: board/OrangePi-Plus2E/overlay/boot/loader.conf [hunk 4]
```diff
@@ -0,0 +1 @@
+fdt_file=orangepi-plus-2e.dtb
```
Comment (evadot) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: Again not needed, the problem is in your uboot port, you must set the MODEL variable to the dtb name (minus ".dtb")

Comment (vgrebenschikov) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: So fdt_file=orangepi-plus-2e.dtb will be compiled into u-boot?

Comment (evadot) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: No, setting MODEL=blah.dtb sets the variable fdtfile into uboot env, which ubldr will import and use as the default dtb to load.
See https://svnweb.freebsd.org/ports/head/sysutils/u-boot-olimex-a20-som-evb/files/patch-include_configs_sunxi-common.h?revision=418446&view=markup#l44

Comment (vgrebenschikov) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: That way there is some disagreement with model and default_config:
MODEL=          orangepi-plus-2e
BOARD_CONFIG=   orangepi_plus2e_defconfig

Comment (evadot) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: That's not a problem

Comment (vgrebenschikov) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: Another question, how to turn on vga console?
I've tried stderr/stdout serial,vga but it does not helps

Comment (evadot) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: VGA ? You mean HDMI right ?
I'm not sure that uboot support H3 HDMI controller yet.

Comment (vgrebenschikov) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: Yes, HDMI
And same for X11?

Comment (vgrebenschikov) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: Under Linux console supported, althouth

Comment (evadot) [inline board/OrangePi-Plus2E/overlay/boot/loader.conf]: We currently do not support the HDMI controller nor the display engine on H3 in FreeBSD

**Comment by kientzle** [comment]
I've merged this so others can try it.  If you need additional changes, please send another Pull Request.

Thanks for your hard work on this!

**Comment by vgrebenschikov** [comment]
Thanks!

**Pull request merged**